storage/leveldb/leveldbSerial.go

```diff
@@ -260,15 +260,7 @@ func (s *SerialDB) Close() error {
 	// (just to close some go routines started as edge cases that would otherwise hang)
 	defer s.closer.Close()
 
-	_ = s.putBatch()
-	s.cancel()
-
-	db := s.makeDbPointerNilReturningLast()
-	if db != nil {
-		return db.Close()
-	}
-
-	return nil
+	return s.prepareClosingUnprotected()
```

Comment:
suggestion to rename to `doClose` for 2 reasons:
1. unprotected is usually called in conjunction to having a mutex and this method should not be called outside that mutex locking/unlocking
2. it does not "prepare" the closing, it actually "execute" the closing

Reply:
done